src/pudl/transform/eiaaeo.py

```diff
@@ -268,64 +360,76 @@ def _decumulate(df: pd.DataFrame, decumulate_columns: list[str]) -> pd.DataFrame
     return renamed_for_pudl
 
 
-def check_totals_add_up(capacity) -> tuple[float, float]:
-    """Check that reported totals match with calculated totals.
-
-    We want to check that:
-
-    * the "total" value is equal to the sum of the non-total values for
-      "cumulative planned/unplanned additions/retirements."
-    * cumulative planned + unplanned additions == cumulative total additions
-    """
-    additions_retirements = [
-        "cumulative_planned_additions",
-        "cumulative_unplanned_additions",
-        "cumulative_retirements",
-    ]
-
-    pk = [
-        "report_year",
-        "model_case_eiaaeo",
-        "region",
-        "projection_year",
-    ]
-    addition_retirement_calculated_totals = (
-        capacity.reset_index()
-        .pipe(
-            lambda df: df.loc[
-                df.dimension.notna() & (df.dimension != "total"),
-                pk + additions_retirements,
-            ]
+@asset(io_manager_key="pudl_io_manager")
+def core_eiaaeo__yearly_projected_electric_sales(
+    raw_eiaaeo__electric_power_projections_regional,
+):
+    """Projected electricity sales by customer class."""
+    # Our series names here only have 3 fields:
+    # 1. "Electricity": broad topic
+    # 2. "Electricity Demand": the variable name
+    # 3. "Commercial", "Industrial": etc. the actual dimension
+    # So we rename some of the extracted columns.
+    #
+    # TODO 2024-05-06 (daz): consider parametrizing the field assignments -
+    # though for tables which have a mix of 3/4 field series, we will always
+    # have to do some munging at the end.
+    sanitized = (
+        filter_enrich_sanitize(
+            raw_df=raw_eiaaeo__electric_power_projections_regional,
+            relevant_series_names=("Electricity : Electricity Demand",),
         )
-        .groupby(pk)
-        .sum()
+        .rename(columns={"variable_name": "dimension", "subtopic": "variable_name"})
+        .pipe(_collect_totals)
     )
+    # 1 BKWh = 1e12 Wh = 1e6 MWh
+    sanitized.loc[sanitized.units == "bkwh", "value"] *= 1e6
```

Comment:
not blocking: if most of these tables have similar units and desired conversions it might be good to either make a lil function that converts any unit that shows up or at least put the unit (key) in a dict with the conversion factor.